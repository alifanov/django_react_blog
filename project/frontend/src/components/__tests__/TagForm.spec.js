import React from "react";
import TagForm from "../forms/tag";

import { shallow, mount } from "../../enzyme";
import { MemoryRouter } from "react-router-dom";

import renderer from "react-test-renderer";

describe(">>> TagForm test", () => {
  let form;
  beforeEach(() => {
    form = shallow(<TagForm />);
  });

  it("+++ render component", () => {
    expect(form.length).toEqual(1);
  });
  it("+++ check elements", () => {
    expect(form.find("Form>FormGroup>Input").length).toEqual(1);
    expect(form.find("Form>div>Button").length).toEqual(1);
  });
  it("+++ check add tag name", () => {
    const component = form.dive();
    component
      .find("Form>FormGroup>Input")
      .at(0)
      .simulate("change", { target: { value: "New tag" } });

    expect(form.state().name).toEqual("New tag");
  });
  it("+++ check submit", () => {
    const mockSubmitFn = jest.fn();
    form = shallow(<TagForm onSubmit={mockSubmitFn} />);
    const component = form.dive();
    component
      .find("Form>FormGroup>Input")
      .at(0)
      .simulate("change", { target: { value: "New tag" } });
    component
      .find("Form>div>Button")
      .at(0)
      .simulate("click");

    expect(form.state().name).toEqual("");
    expect(mockSubmitFn.mock.calls.length).toEqual(1);
  });
});

describe(">>>TagForm --- Snapshot", () => {
  it("+++capturing", () => {
    const renderedValue = renderer
      .create(
        <MemoryRouter>
          <TagForm />
        </MemoryRouter>
      )
      .toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
