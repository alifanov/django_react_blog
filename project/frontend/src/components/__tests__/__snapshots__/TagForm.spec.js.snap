// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>TagForm --- Snapshot +++capturing 1`] = `
<div
  className="jumbotron"
>
  <h4>
    New tag
  </h4>
  <form
    className=""
  >
    <div
      className="position-relative form-group"
    >
      <label
        className=""
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form-control"
        id="name"
        name="name"
        onChange={[Function]}
        placeholder="Enter tag name"
        type="text"
        value=""
      />
    </div>
    <div
      className="d-flex justify-content-end"
    >
      <button
        aria-label={null}
        className="btn btn-secondary btn-sm"
        onClick={[Function]}
        type="button"
      >
        Add new post
      </button>
    </div>
  </form>
</div>
`;
